xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-rest/src/main/java/org/xwiki/livedata/internal/rest/DefaultLiveDataEntryResource.java

```diff
@@ -72,14 +73,14 @@ public Response updateEntry(String sourceId, String namespace, String entryId, E
             String idProperty = config.getMeta().getEntryDescriptor().getIdProperty();
             entry.getValues().put(idProperty, entryId);
             LiveDataEntryStore entryStore = source.get().getEntries();
-            Optional<Object> updatedEntryId = entryStore.save(entry.getValues());
-            if (updatedEntryId.isPresent()) {
-                Optional<Map<String, Object>> values = entryStore.get(updatedEntryId.get());
-                if (values.isPresent()) {
-                    Entry updatedEntry =
-                        createEntry(values.get(), updatedEntryId.get(), config.getQuery().getSource(), namespace);
-                    return Response.status(Status.ACCEPTED).entity(updatedEntry).build();
-                }
+            entryStore.updateAll(entryId, entry.getValues());
```

Comment:
Update the properties of the entry instead of saving a new entry.

Reply:
See above. Save is not just for new entry.